### PR TITLE
fix(): add taskrun values inside workerDirectory task

### LINKED ISSUE #1809: Simplify `EachParallel` with `WorkingDirectory` child tasks
### Issue description

This flow works:

```yaml
id: reproducer
namespace: dev

tasks:
  - id: parallel
    type: io.kestra.core.tasks.flows.EachParallel
    value: ["1", "2", "3", "4", "5", "6"]
    tasks:
      - id: seq
        type: io.kestra.core.tasks.flows.Sequential
        tasks:
        - id: pythonScripts
          type: io.kestra.core.tasks.flows.WorkingDirectory
          tasks:
            - id: cloneRepository
              type: io.kestra.plugin.git.Clone
              url: https://github.com/kestra-io/scripts
              branch: main
            - id: python
              type: io.kestra.plugin.scripts.python.Commands
              runner: DOCKER
              docker:
                image: ghcr.io/kestra-io/pydata:latest
              commands:
                - python etl/parametrized.py {{ parents[0].taskrun.value }}
```

however, it seems that it should already work without having to add the `Sequential` task, because `WorkingDirectory` already runs everything sequentially. However, without Sequential, the `taskrun.value` cannot be accessed:

```yaml
id: reproducer
namespace: dev

tasks:
  - id: parallel
    type: io.kestra.core.tasks.flows.EachParallel
    value: ["1", "2", "3", "4", "5", "6"]
    tasks:
      - id: pythonScripts
        type: io.kestra.core.tasks.flows.WorkingDirectory
        tasks:
          - id: cloneRepository
            type: io.kestra.plugin.git.Clone
            url: https://github.com/kestra-io/scripts
            branch: main
          - id: python
            type: io.kestra.plugin.scripts.python.Commands
            runner: DOCKER
            docker:
              image: ghcr.io/kestra-io/pydata:latest
            commands:
              - python etl/parametrized.py {{ taskrun.value }}
```

this gives - `Missing variable: 'value' on 'python etl/parametrized.py {{ taskrun.value }}' at line 1`

cc @loicmathieu

### PR DESCRIPTION
**Pull request by Skraye**
The issue here is that when in a workerDirectory, we don't follow the usual paths [here](https://github.com/kestra-io/kestra/blob/4a1a8a2515dcf0931c95b2fca57407df46b28a2e/core/src/main/java/io/kestra/core/runners/ExecutorService.java#L226), the parents & parent variable isn't provided

So when we pass in[ this method](https://github.com/kestra-io/kestra/blob/4a1a8a2515dcf0931c95b2fca57407df46b28a2e/core/src/main/java/io/kestra/core/runners/RunContext.java#L403), the variable taskrun is erased.

So, in my solution, when creating children tasks inside the worker, I create a new workerTaskrun with taskRun (maybe useless, but at least the data are here), and copy parents and parent of the workerDirectory.

close #1809

### CODE REVIEW

**Comment by loicmathieu** [comment]
@Skraye the runcontext is created a few lines upper, I wonder if we shouldn't have a methods `forWorkingDirectoryTask` to init it that will do this trick at creation time, this will be more coherent with the other usage.

**Comment by Skraye** [comment]
> @Skraye the runcontext is created a few lines upper, I wonder if we shouldn't have a methods `forWorkingDirectoryTask` to init it that will do this trick at creation time, this will be more coherent with the other usage.

Totally make sense, will change that

File: core/src/main/java/io/kestra/core/runners/RunContext.java [hunk 1]
```diff
@@ -426,6 +426,26 @@ public RunContext forWorker(ApplicationContext applicationContext, WorkerTrigger
...
+            clone.put("parents", clone.get("parents"));
+        }
+        if (clone.containsKey("parent")) {
+            clone.put("parent", clone.get("parent"));
+        }
+
+        this.variables = ImmutableMap.copyOf(clone);
+
+        return this;
+    }
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/runners/RunContext.java:447]: Can you merge those two methods in one ?

We really don't want a method that allow mutation of variables to be public!

**Comment by Skraye** [comment]
> It miss a test, can you check in the existing test that the taskrun.value var is set?

Didn't catch how I could test context, so I did a test that run a flow which used taskrun/parents/parent and checked the content of the parent (the one that contains the value)

**Comment by loicmathieu** [comment]
I didn't saw that you add a `workerTaskRun` and not the standard `taskrun` variable which is not a good idea IMHO.

**Comment by Skraye** [comment]
@loicmathieu 
There is the standard `taskrun`, and the` workerTaskRun`. 
When you have nested flowable, in variables you have a `parents` and a `parent`.
The `parent` is the `taskrun` property of the above level. As we don't follow the standard with the `workingDirectory`, we don't provide the `parents` and the `parent`. 
As the `taskrun` of the workingDirectory does not contain any value and will likely never be used, I decided just to pass the `parents` and `parent` and not adding the workingDirectory inside to avoid adding useless complexity.

This way, when you're in a nested Each, workingDirectory or not its always `parent.taskrun.value`

And finally, just because I had the information, and maybe one day, one user from nowhere will ask for the taskRun value inside a workingDirectory child task, so I put the `workerTaskRun`

**Review by loicmathieu — APPROVED** [review]
LGTM

**Pull request merged**